lib/server.js

```diff
@@ -8,27 +8,36 @@ module.exports = {
 
   createServer (config) {
     const server = new Hapi.Server()
-    server.connection({
+    const defaultOptions = {
       host: config.web.host,
       port: config.web.port,
       routes: {
         files: {
           relativeTo: config.main.paths.root
         }
       }
-    })
+    }
+
+    server.connection(_.defaultsDeep(config.web.options || {}, defaultOptions))
 
     return server
   },
 
   registerPlugins (app, server) {
-    return new Promise((resolve, reject) => {
-      server.register(app.config.web.plugins || [ ], err => {
-        if (err) return reject(err)
+    const promises = _.map(app.config.web.plugins || [], plugin => {
```

Comment:
@konstantinzolotarev I think you shouldn't need to `.map` here.

http://hapijs.com/api#serverregisterplugins-options-callback